201-vmss-automation-dsc/nested/provisionServer.json

```diff
@@ -0,0 +1,254 @@
+                        "TypeName": "System.String"
+                      },
+                      {
+                        "Name": "RebootNodeIfNeeded",
+                        "Value": true,
+                        "TypeName": "System.Boolean"
+                      },
+                      {
+                        "Name": "ActionAfterReboot",
+                        "Value": "ContinueConfiguration",
+                        "TypeName": "System.String"
+                      }
+                    ]
+                  }
+                }
+              }
+            ]
+          },
+          "osProfile": {
+            "computerNamePrefix": "[toLower(parameters('vmssName'))]",
+            "adminUsername": "[parameters('adminUsername')]",
+            "adminPassword": "[parameters('adminPassword')]"
+          },
+          "networkProfile": {
+            "networkInterfaceConfigurations": [
+              {
+                "name": "[parameters('nicName')]",
+                "properties": {
+                  "primary": true,
+                  "ipConfigurations": [
+                    {
+                      "name": "ipConfig",
+                      "properties": {
+                        "subnet": {
+                          "id": "[concat(resourceId('Microsoft.Network/virtualNetworks/',parameters('virtualNetworkName')),'/subnets/',parameters('subnetName'))]"
+                        },
+                        "loadBalancerBackendAddressPools": [
+                          {
+                            "id": "[concat(resourceId('Microsoft.Network/loadBalancers/',parameters('loadBalancerName')),'/backendAddressPools/backendAddressPool')]"
+                          }
+                        ],
+                        "loadBalancerInboundNatPools": [
+                          {
+                            "id": "[concat(resourceId('Microsoft.Network/loadBalancers/',parameters('loadBalancerName')),'/inboundNatPools/natRDPPool')]"
+                          },
+                          {
+                            "id": "[concat(resourceId('Microsoft.Network/loadBalancers/',parameters('loadBalancerName')),'/inboundNatPools/natWinRMPool')]"
+                          }
+                        ]
+                      }
+                    }
+                  ]
+                }
+              }
+            ]
+          }
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Insights/autoscaleSettings",
+      "apiVersion": "2015-04-01",
+      "name": "autoscalehost",
+      "location": "[parameters('location')]",
+      "dependsOn": [
+        "[concat('Microsoft.Compute/virtualMachineScaleSets/', parameters('vmSSName'))]"
+      ],
+      "properties": {
+        "name": "autoscalehost",
+        "targetResourceUri": "[concat('/subscriptions/',subscription().subscriptionId, '/resourceGroups/',  resourceGroup().name, '/providers/Microsoft.Compute/virtualMachineScaleSets/', parameters('vmSSName'))]",
```

Comment:
use resourceID:

resourceId('Microsoft.Compute/virtualMachineScaleSets', parameters('vmSSName'))

Here and below...

Reply:
fixed